docs/content/_docs/config.md

```diff
@@ -0,0 +1,1022 @@
+
+```yaml
+# Local ID for the node.
+id: <string> default = a generated UUID
+
+# When communicating with self, avoid using the network.
+useLocal: <bool> default = false
+
+# The mechanism used to discover nodes in the cluster.
+discovery: <discovery_config>
+
+# Protocols that that node can use to communicate with other Heroic nodes.
+protocols:
+  - <protocol_config>
+  - ...
+
+# Defines a set of tags that identifies which part of the cluster this node belongs to.
+# Nodes that have an identical set of tags are said to be part of the same shard.
+# See the federation section for more details.
+tags:
+  <string>: <string>
+  ...
+
+# Actual topology (shards) is detected based on the metadata coming from the nodes.
+# Expected topology is specified in the optional 'topology' list. This specifies the minimum
+# shards expected, i.e. additional shards may also exist. If an expected shard is not responding the
+# response will include which shard was failing. See the federation section for more details.
+topology:
+  - <string>: <string>
+  - ...
+```
+
+### [`<discovery_config>`](#discovery_config)
+
+The mechanism used to discover nodes in the cluster. Only one disovery type can be set at a time.
+
+#### [static](#static)
+
+Static is the simplest possible form of discovery. It takes a list of nodes that may, or may not be reachable at the moment.
+This list will be queried at a given interval, and any that responds to a metadata request will be added to the local list of known members.
+
+```yaml
+type: static
+
+# The list of nodes that this node will attempt to add to its cluster registry. A valid url has the form protocol://host[:port], eg grpc://localhost:8698
+nodes:
+  - <string>
+  - ...
+```
+
+#### [srv](#srv)
+
+SRV records are useful when running Heroic on more ephemeral environments, like Kubernetes.
+It takes a list of SRV records and resolves each one at a given interval. Any that respond to a metadata
+request will be added to the local list of known members.
+
+```yaml
+type: srv
+
+# The list of srv records that this node will attempt to resolve and add  to its cluster registry.
+records:
+  - <string>
+  - ...
+
+# Protocol to use when building the URI.
+protocol: <string> default = grpc
+
+# Port to use when building the URI. Used to override the port returned by the SRV lookup.
+port: <int> default = result from SRV
+```
+
+### [`<protocol_config>`](#protocol_config)
+
+Protocols that the node can speak to other nodes. Multiple protocols can be enabled at once.
+
+#### [grpc](#grpc)
+
+[gRPC](https://grpc.io) is an open source RPC protocol.
+
+```yaml
+type: grpc
+
+# The address of the interface that this node will bind to.
+host: <string> default = 0.0.0.0
+
+# The port number that this node will bind to.
+port: <int> default = 9698
+
+# Frame size limit in bytes.
+maxFrameSize: <int> default = 10000000
+```
+
+#### [jvm](#jvm)
+
+Communicate directly between multiple JVMs running on the same host.
```

Comment:
this is really only used for testing? maybe make a note of that?

Reply:
Maybe we should deprecate it. The integration tests we run on it are identical to what we run over gRPC.